src/ppom-storage.test.ts

```diff
@@ -0,0 +1,287 @@
+    it('should return data if it matches checksum', async () => {
+      const fileData = {
+        chainId: '1',
+        name: 'dummy',
+        checksum,
+        version: '0',
+      };
+      const ppomStorage = new PPOMStorage({
+        storageBackend: new MockStorageBackend(new ArrayBuffer(123)),
+        readMetadata: () => [fileData],
+        writeMetadata: () => undefined,
+      });
+      const data = await ppomStorage.readFile('dummy', '1');
+      expect(data).toStrictEqual(new ArrayBuffer(123));
+    });
+
+    it('should throw error if checksum does not matches', async () => {
+      const fileData = {
+        chainId: '1',
+        name: 'dummy',
+        version: '0',
+      };
+      const ppomStorage = new PPOMStorage({
+        storageBackend: new MockStorageBackend(new ArrayBuffer(123)),
+        readMetadata: () => [{ ...fileData, checksum: '000' }],
+        writeMetadata: () => undefined,
+      });
+      await expect(async () => {
+        await ppomStorage.readFile('dummy', '1');
+      }).rejects.toThrow('Checksum mismatch');
+    });
+
+    it('should throw error if filemetadata if not found', async () => {
+      const ppomStorage = new PPOMStorage({
+        storageBackend: new MockStorageBackend(new ArrayBuffer(123)),
+        readMetadata: () => [],
+        writeMetadata: () => undefined,
+      });
+      await expect(async () => {
+        await ppomStorage.readFile('dummy', '1');
+      }).rejects.toThrow(
+        'File metadata not found for File (dummy, 1) not found',
+      );
+    });
+
+    it('should throw error if file is not found in storage', async () => {
+      const fileData = {
+        chainId: '1',
+        name: 'dummy',
+        checksum,
+        version: '0',
+      };
+      const ppomStorage = new PPOMStorage({
+        storageBackend: new MockStorageBackend(),
+        readMetadata: () => [fileData],
+        writeMetadata: () => undefined,
+      });
+      await expect(async () => {
+        await ppomStorage.readFile('dummy', '1');
+      }).rejects.toThrow('Storage File (dummy, 1) not found');
+    });
+  });
+
+  describe('PPOMStorage:writeFile', () => {
+    it('should throw error if checksum does not match', async () => {
+      const ppomStorage = new PPOMStorage({
+        storageBackend: new MockStorageBackend(),
+        readMetadata: () => [],
+        writeMetadata: () => undefined,
+      });
+      await expect(async () => {
+        await ppomStorage.writeFile({
+          data: new ArrayBuffer(1),
+          name: 'dummy',
+          chainId: '1',
+          version: '0',
+          checksum,
+        });
+      }).rejects.toThrow('Checksum mismatch');
+    });
+
+    it('should call storageBackend.write if checksum matches', async () => {
+      const mockWrite = jest.fn().mockResolvedValue('test');
+      const ppomStorage = new PPOMStorage({
+        storageBackend: {
+          write: mockWrite,
+        } as unknown as StorageBackend,
```

Comment:
Why is the typecasting needed here? I suspect that you may want to prevent from having to specify all of the required properties for StorageBackend. If that's the case, what do you think about making a `buildStorageBackend` function that returns a complete StorageBackend object but lets you override any properties you want?